Ranaming `master` branch to `main`.

lmarini opened the pull request:
Will just delete master and keep this branch.

lmarini committed:
Ranaming `master` branch to `main`.

robkooper reviewed (changes requested):
`  MASTER_REPO: clowder-framework/clowder`
leave as upper case
`  MAIN_REPO: clowder-framework/clowder`

robkooper commented on doc/src/sphinx/develop/previewers.rst:
```diff
@@ -12,4 +12,4 @@ Previewer can work together with extractors and external services.
 
 Here is a list of previewer embedded with the core
-`source <https://github.com/clowder-framework/clowder/tree/master/public/javascripts/previewers>`_.
+`source <https://github.com/clowder-framework/clowder/tree//public/javascripts/previewers>`_.
```
missing main

robkooper commented on doc/src/sphinx/index.rst:
```diff
@@ -25,7 +25,7 @@ be customized and deployed on your own cloud.
     on `GitHub <https://github.com/clowder-framework>`_.
 
-  * We are always looking for `contributions <https://github.com/clowder-framework/clowder/blob/master/CONTRIBUTING.md>`_.
+  * We are always looking for `contributions <https://github.com/clowder-framework/clowder/blob//CONTRIBUTING.md>`_.
```
missing main

robkooper commented on project/Build.scala:
```diff
@@ -18,7 +18,7 @@ object ApplicationBuild extends Build {
   def appVersion: String = {
     gitBranchName match {
-      case "master" => getVersion
+      case "" => getVersion
```
should this be main?

robkooper commented on public/swagger.yml:
```diff
@@ -16,7 +16,7 @@ info:
     name: The University of Illinois/NCSA Open Source License (NCSA)
-    url: https://github.com/clowder-framework/clowder/blob/master/LICENSE
+    url: https://github.com/clowder-framework/clowder/blob//LICENSE
 
```
should this be main? or only / not //

robkooper commented on release.sh:
```diff
@@ -19,7 +19,7 @@ BRANCH=${BRANCH:-"$(git rev-parse --abbrev-ref HEAD)"}
 
 # find out the version
-if [ "${BRANCH}" = "master" ]; then
+if [ "${BRANCH}" = "" ]; then
```
should this be main?

robkooper commented on release.sh:
```diff
@@ -4,7 +4,7 @@
 
 # can use the following to push to isda-registry for testing:
-# BRANCH="master" SERVER=isda-registry.ncsa.illinois.edu/ ./release.sh
+# BRANCH="" SERVER=isda-registry.ncsa.illinois.edu/ ./release.sh
```
should this be main?

robkooper committed:
Update GitHub actions (#461)

* Ranaming `master` branch to `main`.

* Update GitHub Actions to latest versions

Upgraded various GitHub Actions in ci.yml, docker.yml, and swagger.yml to their latest major versions for improved security and maintenance. Also updated environment variable naming and DockerHub description action usage in docker.yml for consistency and compatibility.

* fix resolvers

small tweak to dockerfile to remove warning

* second reference to as in dockerfile

* another location where typesafe has wrong url

* Remove iRods integration: delete iRods storage service, plugin, config, and docs. Update changelog. Users must use another storage backend (filesystem, MongoDB GridFS, or AWS S3).

* remove typesafe references from jar file

* Fix upload artifact

* more version bumps

* fix python version

---------

Co-authored-by: Luigi Marini <lmarini@illinois.edu>

lmarini committed:
Fix/escape space name description (#459)

* Ranaming `master` branch to `main`.

* Escape space name and description to avoid xss.

* Comment out irods dependency as it is not available anymore.

* Upgrade actions/cache from v1 to v3

robkooper committed:
Fix NoSuchElementException for anonymous users in spaces listing and miniList templates (#460)

- Add user existence checks before accessing user properties in app/views/spaces/listSpaces.scala.html and app/views/spaces/miniList.scala.html
- Update CHANGELOG.md with details of the fix in Unreleased section

Co-authored-by: Luigi Marini <lmarini@illinois.edu>

robkooper committed:
Merge branch 'main' into releasse/v1.23.0

lmarini committed:
Updated version to 1.23.0 and CHANGELOG.md

lmarini committed:
Accidentally checked in commented out services in docker compose

lmarini committed:
Removed ellipses method has it driggers XSS

lmarini committed:
Merge pull request #462 from clowder-framework/releasse/v1.23.0

Release/v1.23.0
